src/server/scriptInfo.ts

```diff
@@ -557,6 +557,8 @@ namespace ts.server {
         }
 
         getLatestVersion() {
+            // Ensure we have updated snapshot to give back latest version
+            this.textStorage.getSnapshot();

```

Comment:
This ensures when isProgramUptodate is checked, we get latest snapshot so the change in the file is reflected correctly (eg. sometimes when reading from file because of watcher, the change in version and text is lazy to ensure we dont change version of scriptshot if file changed but its content didnt - esp happens a lot with file open and close)